components/collector/src/source_collectors/visual_studio_trx/tests.py

```diff
@@ -39,9 +39,10 @@ def _include_entity(self, entity: Entity) -> bool:
     def __entity(test: Element, result: str, namespaces: Namespaces) -> Entity:
         """Transform a test case into a test entity."""
         name = test.attrib["name"]
-        for category in test.findall(".//ns:TestCategoryItem", namespaces):
-            if match := re.search(TestCases.TEST_CASE_KEY_RE, category.attrib["TestCategory"]):
-                name += f" ({match[0]})"
-                break
+        category_items = test.findall(".//ns:TestCategoryItem", namespaces)
+        categories = [item.attrib["TestCategory"] for item in category_items]
+        matches = [match[0] for category in categories if (match := re.search(TestCases.TEST_CASE_KEY_RE, category))]
+        if matches:
+            name += f" ({', '.join(sorted(matches))})"
```

Comment:
Should we deduplicate as well as sorting?

Reply:
If there are duplicates here it means one test has multiple references to the same Jira id. That would be weird, so I'd rather have a chance to detect it.